src/transport_interface/transport_interface_test.c

```diff
@@ -270,12 +300,134 @@ static void prvTransportRecvData( TransportInterface_t * pTransport,
 
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Verify the buffer guard of the test buffer.
+ */
+static void prvVerifyTestBufferGuard( uint8_t * pTransportTestBuffer )
+{
+    /* Prefix and Suffix guard buffers must never change. */
+    TEST_ASSERT_EACH_EQUAL_HEX8_MESSAGE( TRANSPORT_TEST_BUFFER_GUARD_PATTERN,
+                                         pTransportTestBuffer,
+                                         TRANSPORT_TEST_BUFFER_PREFIX_GUARD_LENGTH,
+                                         "transportTestBuffer prefix guard should not be altered." );
+    TEST_ASSERT_EACH_EQUAL_HEX8_MESSAGE( TRANSPORT_TEST_BUFFER_GUARD_PATTERN,
+                                         &( pTransportTestBuffer[ TRANSPORT_TEST_BUFFER_PREFIX_GUARD_LENGTH +
+                                                                  TRANSPORT_TEST_BUFFER_WRITABLE_LENGTH ] ),
+                                         TRANSPORT_TEST_BUFFER_SUFFIX_GUARD_LENGTH,
+                                         "transportTestBuffer suffix guard should not be altered." );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Thread function of the send/receive/compare test.
+ */
+static void prvSendRecvCompareFunc( void * pParam )
+{
+    threadParameter_t * pThreadParameter = pParam;
+    uint8_t * pTrasnportTestBufferStart =
+        &( pThreadParameter->transportTestBuffer[ TRANSPORT_TEST_BUFFER_PREFIX_GUARD_LENGTH ] );
+    uint32_t testBufferSize = 0;
+    NetworkContext_t * pNextworkContext = pThreadParameter->pNetworkContext;
+
+    for( testBufferSize = 1; testBufferSize <= TRANSPORT_TEST_BUFFER_WRITABLE_LENGTH; testBufferSize = testBufferSize + 1 )
+    {
+        /* Initialize the test data buffer. */
+        prvInitializeTestData( pTrasnportTestBufferStart, testBufferSize );
+
+        /* Send the test data to the server. */
+        prvTransportSendData( pTestTransport, pNextworkContext, pTrasnportTestBufferStart,
+                              testBufferSize );
+
+        /* Receive the test data from server. */
+        prvTransportRecvData( pTestTransport, pNextworkContext, pTrasnportTestBufferStart,
+                              testBufferSize );
+
+        /* Compare the test data received from server. */
+        prvVerifyTestData( pTrasnportTestBufferStart, testBufferSize, TRANSPORT_TEST_BUFFER_WRITABLE_LENGTH );
+
+        if( pThreadParameter->stopFlag == true )
+        {
+            break;
+        }
+
+        /* Output information to indicate the test is running. */
+        UNITY_OUTPUT_CHAR( '.' );

```

Comment:
IDT parses the test output in the format of "TEST(group, test_name) PASS/FAIL". Are these dots going to appear in betwen "TEST(group, test_name)" and  "PASS/FAIL"? This can potentially impact IDT's result parsing.

Reply:
As we discuss, we can add a debug macro and default is disabled. Developer can override the macro in test_param_config.h if they want to know the test is still running.